### PR TITLE
add transaction signing API and types, offline signer

### PR DESCRIPTION
**Pull request by ryankurte**
adds `mc-transaction-signer` crate, defining types for communication with view-only accounts (offline, ledger, etc.), as well as helpers for implementing these consistently and the offline signer implementation.

### Motivation

this API doesn't really overlap with the frontend JSON or backend GRPC/proto, but needs to be specified well to support different backends. providing this as a module with types and _super prescriptive) helpers both simplifies other implementations and helps keep these in sync.

blocked on:
- [x] https://github.com/mobilecoinfoundation/mobilecoin/pull/2901 for `ViewAccount` and `TxOutPublicKey` types

replaces: 
- https://github.com/mobilecoinfoundation/mobilecoin/pull/2902

### CODE REVIEW

**Review by briancorbin — APPROVED** [review]
This looks great! Thanks so much for migrating this and refactoring it from Full Service :)

File: transaction/signer/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,319 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Transaction signer types, used for communication with external signer
+//! implementations
```
Comment (cbeck88) [inline transaction/signer/src/lib.rs:4]: This description is rather sparse, can we try to add some description of what an external signer does, or maybe talk about an example?

File: transaction/signer/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+[package]
+name = "mc-transaction-signer"
+version = "4.0.1"
+authors = ["MobileCoin"]
+edition = "2021"
+readme = "README.md"
```
Comment (cbeck88) [inline transaction/signer/Cargo.toml:6]: it looks that there is no README.md, can we add one?

**Review by cbeck88 — APPROVED** [review]
looks good, my only ask is for some more high level docu for this crate

**Pull request merged**